homeassistant/components/zha/core/gateway.py

```diff
@@ -733,6 +733,8 @@ async def shutdown(self) -> None:
         _LOGGER.debug("Shutting down ZHA ControllerApplication")
         for unsubscribe in self._unsubs:
             unsubscribe()
+        for device in self.devices.values():
+            device.async_cleanup_handles()
```

Comment:
This does not resolve all lingering timers, but it does resolve quite a few of them and it will help identify more clearly the remaining ones.